apis/gateway/v2alpha1/jwt_conversion.go

```diff
@@ -37,16 +37,23 @@ func convertOryJwtAccessStrategy(accessStrategy *v1beta1.Authenticator) (*v1beta
 func convertIstioJwtAccessStrategy(accessStrategy *v1beta1.Authenticator) (*v1beta1.JwtConfig, error) {
 	var jwtConfig *v1beta1.JwtConfig
 
+	convertedFromObject := false
 	if accessStrategy.Config.Object != nil {
-		jwtConfig = accessStrategy.Config.Object.(*v1beta1.JwtConfig)
-	} else if accessStrategy.Config.Raw != nil {
+		jwtConfig, convertedFromObject = accessStrategy.Config.Object.(*v1beta1.JwtConfig)
+	}
+
+	if accessStrategy.Config.Raw != nil && !convertedFromObject {
 		jwtConfig = &v1beta1.JwtConfig{}
 		err := json.Unmarshal(accessStrategy.Config.Raw, jwtConfig)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	if jwtConfig == nil {
```

Comment:
The logic that is implemented right now doesn't assume that the configurations are mutually exclusive, but the previous implementation did. Unfortunately, I also cannot see a new test case to validate that new behaviour so I don't know if it was done unintentionally.
I'd not update this implementation without feedback about this new logic.

Reply:
I found out some conversion tests were failing when this was not implemented as it is done here since the `accessStrategy.Config.Object` was not `v1beta1` but `v2alpha1`. It is not mutually exclusive, as in case the typecast fails, we still need to convert the object.

Reply:
That is, the new `to-from` conversion tests are failing without this safeguard